src/components/textarea/textarea.tsx

```diff
@@ -47,24 +51,66 @@ export interface TextAreaProps extends RestrictedTextareaHTMLAttributes {
  */
 export const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
   ({ className, defaultValue, hasError, maxRows = Infinity, minRows = 2, onChange, rows, value, ...rest }, ref) => {
```

Comment:
**note:** Here I am wiring up the new hooks with the primary text area and the newly added shadow text area. Important things to note include:

- `shouldResizeViaJS` determines whether the CSS field-sizing property is supported and if so, we will not do any JS resizing. Also, we only resize via JS when `rows` is undefined.
- Both hooks have an `isEnabled` param that effectively "turns off" the resizing logic they otherwise facilitate. We have to do it this way because we can't conditionally use hooks.
- We also use `shouldResizeViaJS` to determine if we render the shadow text area or not.